streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -719,13 +720,10 @@
         tempConsumerDefaultOverrides.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
         tempConsumerDefaultOverrides.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
         tempConsumerDefaultOverrides.put("internal.leave.group.on.close", false);
-        // MAX_POLL_INTERVAL_MS_CONFIG needs to be large for streams to handle cases when
-        // streams is recovering data from state stores. We may set it to Integer.MAX_VALUE since
-        // the streams code itself catches most exceptions and acts accordingly without needing
-        // this timeout. Note however that deadlocks are not detected (by definition) so we
-        // are losing the ability to detect them by setting this value to large. Hopefully
-        // deadlocks happen very rarely or never.
-        tempConsumerDefaultOverrides.put(ConsumerConfig.MAX_POLL_INTERVAL_MS_CONFIG, Integer.toString(Integer.MAX_VALUE));
+        tempConsumerDefaultOverrides.put(
+            ConsumerConfig.MAX_POLL_INTERVAL_MS_CONFIG,
+            Long.toString(Duration.ofMinutes(1).toMillis())
```

Comment:
`ConsumerConfigs.MAX_POLL_INTERVAL_MS_CONFIG` default value is 5 minutes, so I think it is not necessary to further decrease it to 1 minute in Streams; maybe we can just remove this override to defer to Consumer's default value?

BTW we should have a tiny KIP for this default value change.

Reply:
:+1:

Reply:
Ok, I've taken your recommendation and also created KIP-442: https://cwiki.apache.org/confluence/display/KAFKA/KIP-442%3A+Return+to+default+max+poll+interval+in+Streams (https://cwiki.apache.org/confluence/x/1COGBg)